force-app/main/default/classes/SOQL.cls

```diff
@@ -16,343 +16,463 @@
+        }
+    }
+
+    /**
+     * Question 2
+     * Returns all contacts with first name of John
+     * @return List<Contact> A list of Contact records
+     */
+    public static List<Contact> getContactsByFirstName() {
+        try {
+            return [SELECT Id, FirstName, LastName, Name
+                    FROM Contact
+                    WHERE FirstName = 'John'
+                    AND LastName != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Contact>();
+        }
+    }
+
+    /**
+     * Question 3
+     * Returns Contact Ids for contacts with specific last names
+     * @return List<Id> A list of Contact Ids
+     */
+    public static List<Id> getContactIdsByLastName() {
+        try {
+            List<Id> contactIds = new List<Id>();
+            for(Contact c : [SELECT Id
+                           FROM Contact
+                           WHERE LastName IN ('Smith', 'Jones')
+                           AND LastName != null
+                           LIMIT 50000]) {
+                contactIds.add(c.Id);
+            }
+            return contactIds;
+        } catch(Exception e) {
+            return new List<Id>();
+        }
+    }
+
+    /**
+     * Question 4
+     * Returns List of closed won opportunities for a specific Account
+     * @param accountId The ID of the Account
+     * @return List<Opportunity> List of filtered opportunities
+     */
+    public static List<Opportunity> getClosedWonOpportunityCountByAccount(Id accountId) {
+        try {
+            if(accountId == null) {
+                return new List<Opportunity>();
+            }
+            return [SELECT Id, Name, Amount, CloseDate, AccountId, stageName
+                    FROM Opportunity
+                    WHERE AccountId = :accountId
+                    AND StageName = 'Closed Won'
+                    AND Amount != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Opportunity>();
+        }
+    }
+
+    /**
+     * Question 5
+     * Returns the total count of Account records
+     * @return Integer The count of Account records
+     */
+    public static Integer getAccountCount() {
+        try {
+            Integer count = [SELECT COUNT() FROM Account WHERE Name != null];
+            return count != null ? count : 0;
+        } catch(Exception e) {
+            return 0;
+        }
+    }
+
+    /**
+     * Question 6
+     * Returns tasks due within next 7 days
+     * @return Map<Id, Task> Map of Task Ids to Task records
+     */
+    public static Map<Id, Task> getTasksDueNextWeek() {
+        try {
+            Date today = Date.today();
+            Date nextWeek = today.addDays(7);
+
+            List<Task> tasks = [SELECT Id, Subject, ActivityDate, Status
+                               FROM Task
+                               WHERE ActivityDate >= :today
+                               AND ActivityDate <= :nextWeek
```

Comment:
For practice, try to a date literal  https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_dateformats.htm